src/main/java/uk/gov/hmcts/darts/authorisation/service/impl/AuthorisationServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity_;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class AuthorisationServiceImpl implements AuthorisationService {
+
+    @Autowired
+    private EntityManager em;
+
+    @Override
+    public UserState getAuthorisation(String emailAddress) {
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<GetAuthorisationResult> criteriaQuery = criteriaBuilder.createQuery(GetAuthorisationResult.class);
+
+        Root<UserAccountEntity> root = criteriaQuery.from(UserAccountEntity.class);
+        Join<UserAccountEntity, SecurityGroupEntity> securityGroup = root.join(UserAccountEntity_.securityGroupEntities);
+        Join<SecurityGroupEntity, SecurityRoleEntity> securityRole = securityGroup.join(
+            SecurityGroupEntity_.securityRoleId);
+        Join<SecurityRoleEntity, SecurityPermissionEntity> securityPermission = securityRole.join(
+            SecurityRoleEntity_.securityPermissionEntities);
+
+        criteriaQuery.select(criteriaBuilder.construct(
+                                 GetAuthorisationResult.class,
+                                 root.get(UserAccountEntity_.id),
+                                 root.get(UserAccountEntity_.username),
+                                 securityRole.get(SecurityRoleEntity_.id),
+                                 securityRole.get(SecurityRoleEntity_.roleName),
+                                 securityPermission.get(SecurityPermissionEntity_.id),
+                                 securityPermission.get(SecurityPermissionEntity_.permissionName)
+                             )
+        );
+
+        ParameterExpression<String> paramEmailAddress = criteriaBuilder.parameter(String.class);
+        criteriaQuery.where(criteriaBuilder.equal(
+            criteriaBuilder.lower(root.get(UserAccountEntity_.emailAddress)),
+            criteriaBuilder.lower(paramEmailAddress)
+        ));
+
+        TypedQuery<GetAuthorisationResult> query = em.createQuery(criteriaQuery);
+        query.setParameter(paramEmailAddress, emailAddress);
+
+        return getUserState(query.getResultList());
+    }
+
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    private UserState getUserState(List<GetAuthorisationResult> getAuthorisationResultList) {
+        UserState.UserStateBuilder userStateBuilder = UserState.builder();
+        Integer tmpRoleId = 0;
+        Set<Role> roles = new LinkedHashSet<>();
+        Set<Permission> permissions = new LinkedHashSet<>();
+
+        for (GetAuthorisationResult result : getAuthorisationResultList) {
+            userStateBuilder.userId(result.userId());
+            userStateBuilder.userName(result.userName());
+            userStateBuilder.roles(roles);
+
+            if (!tmpRoleId.equals(result.roleId())) {
```

Comment:
Yes, I think there can be because our security database model goes through tables: user_account - security_group (ManyToMany) and security_group - security_role (ManyToOne) and security_role - security_permission (ManyToMany)
[modernised_darts_erd](https://github.com/hmcts/darts-api/blob/master/src/main/resources/db/reference/modernised_darts_erd.pdf)

I've tried to cover this scenario in an integration test for a user ("test.bristol@example.com") with 2 security groups - see Test shouldGetAuthorisationForTestBristol

- https://github.com/hmcts/darts-api/blob/DMP-637-authorisation-service/src/integrationTest/java/uk/gov/hmcts/darts/authorisation/service/AuthorisationServiceTest.java#L58 
- https://github.com/hmcts/darts-api/blob/DMP-637-authorisation-service/src/integrationTest/java/uk/gov/hmcts/darts/authorisation/service/AuthorisationServiceTest.java#L63C10-L81